shells/backends/gcp_tf_backend/src/driver.py

```diff
@@ -0,0 +1,295 @@
+                        # logger.exception(f"def: {json}")
+                        os.environ[GOOGLE_APPLICATION_CREDENTIALS] = clp_json_path
+                        # self._handle_exception_logging(logger, f"CLP: {clp_json_path}  DEF: {json}") 
+                        # os.environ[GOOGLE_APPLICATION_CREDENTIALS] = "C:\\Users\\CSadmin123456\\CS\\tf-gcp\\prod.json"
+            except Exception as e:
+                self._handle_exception_logging(logger, f"There was an issue accessing GCP. {e}")
+            bucket_name = gcp_backend_resource.bucket_name
+            if bucket_name:      
+                self._validate_bucket_exists(bucket_name, context)
+
+    def _get_attrbiute_value_from_clp(self, attributes, model_prefix, attribute_name) -> str:
+        for attribute in attributes:
+
+            if attribute.Name == f"{model_prefix}{attribute_name}":
+                return attribute.Value
+        return ""
+
+
+    def _fill_backend_sercret_vars_data(self, myactual, clp):
+        print(myactual)
+        for attr in  myactual.ResourceAttributes:
+            if attr.Name == "Google Cloud Provider.Credentials Json Path":
+                clp_json_path = attr.Value
+        # gcp_model_prefix = ""
+        # self._backend_secret_vars = {}
+        # for attr in clp_resource_attributes:
+        #     if attr.Name == gcp_model_prefix + "CREDENTIALS JSON PATH":
+        #         json_path_new = attr.Value
+                return clp_json_path
+                # self._backend_secret_vars["subscription_id"] = attr.Value
+            # if attr.Name == azure_model_prefix + "Azure Tenant ID":
+            #     self._backend_secret_vars["tenant_id"] = attr.Value
+            # if attr.Name == azure_model_prefix + "Azure Application ID":
+            #     self._backend_secret_vars["client_id"] = attr.Value
+            # if attr.Name == azure_model_prefix + "Azure Application Key":
+            #     dec_client_secret = api.DecryptPassword(attr.Value).Value
+            #     self._backend_secret_vars["client_secret"] = dec_client_secret   
+
+    # def _create_clp_gcp_session(self, json_path):
+    #     os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = json_path
+
+
+    def _validate_clp(self, api, gcp_backend_resource, logger):
+        clp_resource_name = gcp_backend_resource.cloud_provider
+        clp_details = api.GetResourceDetails(clp_resource_name)
+        clp_res_model = clp_details.ResourceModelName
+        clpr_res_fam = clp_details.ResourceFamilyName
+        if (clpr_res_fam != 'Cloud Provider' and clpr_res_fam != 'CS_CloudProvider') or \
+                clp_res_model not in GCP_MODELS:
+            logger.error(f"Cloud Provider does not have the expected type: {clpr_res_fam}")
+            raise ValueError(f"Cloud Provider does not have the expected type:{clpr_res_fam}")
+        return clp_details
+
+    def _handle_exception_logging(self, logger, msg):
```

Comment:
this method raises an exception it not only handles logging
the name is a little bit confusing
I would call this something like "raise_and_log"